src/main.rs

```diff
@@ -532,9 +617,38 @@ fn validate_body(
 
     let mut sha256 = Sha256::new();
     sha256.update(response_body);
-    let body_sha = sha256.finalize();
+    let body_sha: [u8; 32] = sha256.finalize().into();
+
+    if let Some(tree) = chunk_tree {
+        let chunk_tree: HashTree = serde_cbor::from_slice(&tree).map_err(AgentError::InvalidCborData)?;
 
-    Ok(&body_sha[..] == tree_sha)
+        let chunk_tree_digest = chunk_tree.digest();
+
+        if chunk_tree_digest != tree_sha {
+            slog::trace!(
```

Comment:
Maybe this is a warning not just trace?

Reply:
Same in 639 below.

Reply:
A similar design is already used nearby. If it's really necessary, then I'll change it :)